symbulate/plot.py

```diff
@@ -40,4 +41,108 @@ def plot(*args, **kwargs):
     except:
         plt.plot(*args, **kwargs)
         
+def is_discrete(heights):
+    return sum([(i > 1) for i in heights]) > .8 * len(heights)
+
+def count_var(x):
+    counts = {}
+    for val in x:
+        if val in counts:
+            counts[val] += 1
+        else:
+            counts[val] = 1
+    return counts
+    
+def compute_density(values):
+	density = gaussian_kde(values)
+	density.covariance_factor = lambda: 0.25
+	density._compute_covariance()
+	return density
+
+def setup_ticks(pos, lab, ax, axis):
+    if axis == 'x':
+        ax.set_xticks(pos)
+        ax.set_xticklabels(lab)
+    elif axis == 'y':
+        ax.set_yticks(pos)
+        ax.set_yticklabels(lab)
+    
+def add_colorbar(fig, type, mappable, label):
+    if 'marginal' not in type:
+        caxes = fig.add_axes([0, 0.1, 0.05, 0.8])
+    else:
+        caxes = fig.add_axes([0, 0.1, 0.05, 0.57])
+    cbar = plt.colorbar(mappable=mappable, cax=caxes)
+    caxes.yaxis.set_ticks_position('left')
+    cbar.set_label(label)
+    caxes.yaxis.set_label_position('left')
+    return cbar, caxes
+
+def setup_tile(v, bins, discrete):
+    if not discrete:
+        v_lab = np.linspace(min(v), max(v) + 1, bins)
+        v_pos = range(len(v_lab))
+        v_vect = np.digitize(v, v_lab)
```

Comment:
I'm pretty sure that the plots that this code will produce are off by 1. Can you please double check this?

The lowest number when you call `np.digitize` is going to be 1 (because every value in `v` is >= min(v)). But `v_pos` starts at 0. Am I missing something here?

Reply:
For clarification, do you mean that the labels are shifted over by 1 or that the actual positions within the intensity matrix will be shifted over by 1?

Reply:
It doesn't matter if the labels and intensity matrix are **both** shifted over by 1. The problem is that they're inconsistent with each other.

As @kevindavisross pointed out, the labels are still at the midpoints of the tiles. They should be at the endpoints of the tiles, and you should double check that the numbers are correct (i.e., not shifted over by 1) by plotting a few examples where you know what the plot should look like.